src/server/subclient.cpp

```diff
@@ -82,7 +82,14 @@ namespace aperf {
 
     if (!last_block) {
       recurse(*elem, callchain_parts, callchain_index + 1, period,
-              time_ordered, offcpu);
+              time_ordered, offcpu, timestamp);
+    }else{
+      if (!elem->contains("samples")) {
+          (*elem)["samples"] = nlohmann::json::array();
+      }
+      (*elem)["samples"].push_back(nlohmann::json({
+      {"timestamp", timestamp},
+      {"period",period}}));
     }
```

Comment:
It's unlikely for samples to be needed for non-time-ordered flame graphs. Try updating the code so that time-ordered flame graphs have samples and non-time-ordered ones don't.

Reply:
Done :+1: